src/gucs/mod.rs

```diff
@@ -7,5 +12,20 @@ pub unsafe fn init() {
         self::planning::init();
         self::internal::init();
         self::executing::init();
+        self::embedding::init();
+    }
+}
+
+fn guc_string_parse(
```

Comment:
There is a GUC check hook to check if the string is valid UTF-8 string. Let's use that.